src/lib/core/components/visualizations/implementations/HistogramVisualization.tsx

```diff
@@ -846,11 +854,40 @@ function HistogramViz(props: VisualizationProps<Options>) {
         <div style={{ display: 'flex', flexDirection: 'column' }}>
           {/* make switch and radiobutton single line with space
                   also marginRight at LabelledGroup is set to 0.5625em: default - 1.5625em*/}
-          <LabelledGroup
-            label="X-axis controls"
-            containerStyles={{
-              marginRight: '1em',
-              // marginRight: valueType === 'date' ? '1em' : '0em',
+
+          {/* set Undo icon and its behavior */}
+          <div
+            style={{
+              display: 'flex',
+              flexDirection: 'row',
+              alignItems: 'center',
+            }}
+          >
+            <LabelledGroup label="X-axis controls"> </LabelledGroup>
```

Comment:
I see LabelledGroup is now childless :cry: 

Does it cause problems if we put all the controls inside it?

Reply:
yeah it is now childless... the reason why I separate it is that I need to place label and reset button (Undo icon) in a row, so I wrapped <LabelledGroup> and reset button with <div>.